### ISSUE TITLE
Problem with claiming unnecessary generic parameter for types.Union

### ISSUE THREAD

**Issue by mano7onam** [microsoft/pyright#1893]
**Describe the bug**
The problem:
```
/home/runner/work/typeshed/typeshed/stdlib/types.pyi
  /home/runner/work/typeshed/typeshed/stdlib/types.pyi:355:39 - error: Expected type arguments for generic class "Union" (reportMissingTypeArgument)
  /home/runner/work/typeshed/typeshed/stdlib/types.pyi:356:40 - error: Expected type arguments for generic class "Union" (reportMissingTypeArgument)
```

for the code part:
```
class Union:
        __args__: Tuple[Any, ...]
        def __or__(self, obj: Any) -> Union: ...
        def __ror__(self, obj: Any) -> Union: ...
```
but Union is not ```typing.Union```. 
It is in ```types.pyi``` and it's ```types.Union```

The problem is [here](https://github.com/microsoft/pyright/blob/9848f0daf6aa54a2829cbc6e6b64ecfd46c33db1/packages/pyright-internal/src/analyzer/typeUtils.ts#L2068) and [call is here](https://github.com/microsoft/pyright/blob/9848f0daf6aa54a2829cbc6e6b64ecfd46c33db1/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L3847)

**To Reproduce**
Run CI tests for the [pull request](https://github.com/python/typeshed/pull/5509)

**Expected behavior**
Checks should pass

**Comment by erictraut** [comment]
Apologies for the bug, and thanks for reporting it.

Pyright allows special-form symbols to be defined in a handful of core typeshed stubs (including "builtins.pyi, typing.pyi, typing_extensions.pyi, and types.pyi). This accommodates types that are defined in both typing and typing_extensions, for example. In this particular case, it was in appropriately assuming that `Union` was the special form defined in `typing.pyi`.

I've checked in a fix, and I can publish an updated version of pyright shortly to unblock your work on typeshed.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.142, which I just published. It will also be included in the next release of pylance.

@mano7onam, would you like to increment the version of pyright in your PR? That's the fastest way to unblock you. You need to change the version in two places: in tests/pyright_test.py and .github/workflows/tests.yml. It's currently set to "1.1.140" and needs to be updated to "1.1.142".

Alternatively, we could submit a separate PR to typeshed that bumps the version, and you could wait for that PR to get merged. Whatever works best for you.

**Comment by jakebailey** [comment]
I sent https://github.com/python/typeshed/pull/5515 already (oops); if they bump it in their PR we can confirm the fix instead and do it there, no worries.
